pkg/cli/debug_job_trace.go

```diff
@@ -73,7 +81,91 @@ func getJobTraceID(sqlConn *sqlConn, jobID int64) (int64, error) {
 	return traceID, nil
 }
 
-func writeTracePayloadJSON(sqlConn *sqlConn, jobID int64, traceFilePath string) error {
+type inflightTraceRow struct {
+	nodeID     int64
+	rootOpName string
+	traceJSON  string
+	traceStr   string
+	jaegerJSON string
+}
+
+// stitchJaegerJSON adds the trace spans from jaegerJSON into the
+// nodeTraceCollection object, and returns a new cumulative
+// tracing.TraceCollection object.
+func stitchJaegerJSON(
+	nodeTraceCollection *tracing.TraceCollection, jaegerJSON string,
+) (*tracing.TraceCollection, error) {
+	var cumulativeTraceCollection *tracing.TraceCollection
+
+	// Unmarshal the jaegerJSON string to a TraceCollection.
+	var curTraceCollection tracing.TraceCollection
+	if err := json.Unmarshal([]byte(jaegerJSON), &curTraceCollection); err != nil {
+		return cumulativeTraceCollection, err
+	}
+
+	// Sanity check that the TraceCollection has a single trace entry.
+	if len(curTraceCollection.Data) != 1 {
+		return cumulativeTraceCollection, errors.AssertionFailedf("expected a single trace but found %d",
+			len(curTraceCollection.Data))
+	}
+
+	if nodeTraceCollection == nil {
+		cumulativeTraceCollection = &curTraceCollection
+		return cumulativeTraceCollection, nil
+	}
+	cumulativeTraceCollection = nodeTraceCollection
+
+	// Sanity check that the TraceID of the new and cumulative TraceCollections is
+	// the same.
+	if cumulativeTraceCollection.Data[0].TraceID != curTraceCollection.Data[0].TraceID {
+		return cumulativeTraceCollection, errors.AssertionFailedf("expected traceID of nodeTrace: %d and curTrace: %d to be equal",
+			cumulativeTraceCollection.Data[0].TraceID, curTraceCollection.Data[0].TraceID)
+	}
+
+	// Add spans from the curTraceCollection to the nodeTraceCollection.
+	for _, span := range curTraceCollection.Data[0].Spans {
+		cumulativeTraceCollection.Data[0].Spans = append(cumulativeTraceCollection.Data[0].Spans, span)
+	}
+
+	return cumulativeTraceCollection, nil
+}
+
+func populateInflightTraceRow(vals []driver.Value) (inflightTraceRow, error) {
+	var row inflightTraceRow
+	if id, ok := vals[0].(int64); ok {
+		row.nodeID = id
+
+	} else {
+		return row, errors.Errorf("unexpected value: %T of %v", vals[0], vals[0])
+	}
+
+	if rootOpName, ok := vals[1].(string); ok {
+		row.rootOpName = rootOpName
+	} else {
+		return row, errors.Errorf("unexpected value: %T of %v", vals[1], vals[1])
+	}
+
+	if traceJSON, ok := vals[2].(string); ok {
+		row.traceJSON = traceJSON
+	} else {
+		return row, errors.Errorf("unexpected value: %T of %v", vals[1], vals[1])
```

Comment:
These indexes are off by 1 (same with the checks below I think)

Reply:
yup, fixed.